clang-tools-extra/clang-doc/tool/CMakeLists.txt

```diff
@@ -25,3 +25,11 @@ install(FILES ../assets/clang-doc-default-stylesheet.css
 install(FILES ../assets/index.js
   DESTINATION "${CMAKE_INSTALL_DATADIR}/clang"
   COMPONENT clang-doc)
+
+add_custom_target(copy-clang-doc-assets
```

Comment:
We prefer having a rule per file since most systems don't have a reliable way to tell if the directory content has changed. The approach we usually use to to have one `add_custom_command` per file with a single `add_custom_target` that depends on all output files, see for example https://github.com/llvm/llvm-project/blob/092dbfaad257885692fa64559e9eb43a5c466798/libcxx/include/CMakeLists.txt#L1024-L1043 or https://github.com/llvm/llvm-project/blob/092dbfaad257885692fa64559e9eb43a5c466798/clang/lib/Headers/CMakeLists.txt

Reply:
OK, I’ll take a pass at adapting one of those for use here. If I think I can generalize it, I’ll make another PR to add it as a utility in the llvm cmake module.

Reply:
I think this is addressed now. I more or less cribed directly from the clang header example. Initially I tried to have the function be a bit more generic and put it in `llvm/cmake/modules/CopyLLVMFiles.cmake`, but my normal methods of including don't seem to work in `clang-doc`. Likely I'm just holding that part wrong, but I don't see anything in `clang-tools-extra` using `include(AddLLVM)` or some other module. 

Happy to refactor if you have some other suggestions.